code/DMSDocumentCartExtension.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+/**
+ * Class DMSDocumentCartExtension adds a control
+ *
+ * @property Boolean     AllowedInCart
+ *
+ * @property DMSDocument owner
+ */
+class DMSDocumentCartExtension extends DataExtension
+{
+    private static $db = array(
+        'AllowedInCart' => 'Boolean',
+    );
+
+    public function isAllowedInCart()
+    {
+        return $this->owner->AllowedInCart;
+    }
+
+    /**
+     * @param FieldList $fields
+     */
+    public function updateCMSFields(FieldList $fields)
+    {
+        $fields->insertBefore(
+            CheckboxField::create(
+                'AllowedInCart',
+                _t('DMSDocument.ALLOWED_IN_CART', 'Allowed in document cart')
```

Comment:
Translations should be belong to their respective class or modules, so this should really be `DMSDocumentCart.ALLOWED_IN_CART` rather than `DMSDocument`